x-pack/legacy/plugins/maps/public/angular/map_controller.js

```diff
@@ -174,26 +174,27 @@ app.controller('GisMapController', ($scope, $route, kbnUrl, localStorage, AppSta
     }
   }
 
-  $scope.$watch('savedQuery', (newSavedQuery, oldSavedQuery) => {
+  $scope.$watch('savedQuery', (newSavedQuery) => {
     if (!newSavedQuery) return;
 
     $state.savedQuery = newSavedQuery.id;
-
-    if (newSavedQuery.id === (oldSavedQuery && oldSavedQuery.id)) {
-      updateStateFromSavedQuery(newSavedQuery);
-    }
+    updateStateFromSavedQuery(newSavedQuery);
   });
 
   $scope.$watch(() => $state.savedQuery, newSavedQueryId => {
     if (!newSavedQueryId) {
       $scope.savedQuery = undefined;
       return;
     }
-    savedQueryService.getSavedQuery(newSavedQueryId).then((savedQuery) => {
-      $scope.$evalAsync(() => {
-        $scope.savedQuery = savedQuery;
+
+    if ($scope.savedQuery && newSavedQueryId !== $scope.savedQuery.id) {
```

Comment:
This will now only run on page load, and when someone manually changes the ID in the URL.